adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,40 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <algorithm>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isCloseBracket(const char bracket) {
+    auto resultIter = std::find_if(brackets.begin(), brackets.end(), [bracket](const std::pair<char, char> value) {
+        return value.second == bracket;
+    });
+    return resultIter != brackets.end();
+}
+
+bool isValid(const std::string& source) {
+    std::stack<char> bracketsStack;
+
+    for (const char &ch : source) {
+        auto resultIter = std::find_if(brackets.begin(), brackets.end(), [ch](const std::pair<char, char> value) {
+            return value.first == ch;
+        });
+
+        if (resultIter != brackets.end()) {
+            bracketsStack.push((*resultIter).second);
+        } else if (isCloseBracket(ch)) {
+            if (bracketsStack.empty()) {
```

Comment:
you should check stack top value here too

Reply:
I don't understand what do you mean.

Reply:
I mean the case when stack top value (open bracket) has another type

Reply:
"another type" is it mean other symbol? if other symbol I skip it and add only brackets to the stack or I something don't understand

Reply:
See this example:
[{]}
Stack:
[
{
Then, next symbol is "]" - so, bracket on top of the stack ("}") is not the same type. You can return immediately.

Reply:
done. thanks